app/src/main/java/xyz/wim/dentrealitytask/ui/map/MapFragment.kt

```diff
@@ -1,107 +1,139 @@
+import kotlinx.coroutines.launch
 import org.koin.androidx.viewmodel.ext.android.viewModel
 import timber.log.Timber
 import xyz.wim.dentrealitytask.R
-import xyz.wim.dentrealitytask.data.Country
-import xyz.wim.dentrealitytask.databinding.DetailsFragmentBinding
+import xyz.wim.dentrealitytask.await
 import xyz.wim.dentrealitytask.databinding.MapFragmentBinding
 import xyz.wim.dentrealitytask.ui.details.DetailsFragment
-import kotlin.math.roundToInt
 
 
 class MapFragment : Fragment() {
 
     companion object {
         fun newInstance() = MapFragment()
-    }
 
-    private var root: View? = null
+        private const val MAPVIEW_BUNDLE_KEY = "MapViewBundleKey"
+    }
 
     private var _binding: MapFragmentBinding? = null
 
-    // This property is only valid between onCreateView and
-// onDestroyView.
+    // This property is only valid between onCreateView and onDestroyView.
     private val binding get() = _binding!!
 
     private val vm: MapViewModel by viewModel()
 
+    private val mapChannel = Channel<GoogleMap>()
+
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
+    ): View {
         _binding = MapFragmentBinding.inflate(inflater, container, false)
-        val view = _binding!!.root
-        return view
+        lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                displayCountryMarkers()
+            }
+        }
+        with(binding) {
+            mapView.onCreate(savedInstanceState)
+            mapView.getMapAsync {
+                lifecycleScope.launch {
+                    mapChannel.send(it)
+                }
+            }
+        }
+
+        return _binding!!.root
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         vm.countries.observe(this) {
             Timber.d("Observing countries")
-            createCountryViews(it)
         }
         super.onCreate(savedInstanceState)
     }
 
-    override fun onDestroyView() {
-        super.onDestroyView()
-        _binding = null
-    }
+    private suspend fun displayCountryMarkers() {
+        val map = mapChannel.receive()
+        val countries = vm.countries.await()
 
-    private fun createCountryViews(countries: List<Country>) {
         for (country in countries) {
-            createCountryView(country)
+            val latlng = LatLng(country.latlng[0], country.latlng[1])
+            map.addMarker(
+                MarkerOptions()
+                    .position(latlng)
+                    .title(country.countryCode)
+            )
+        }
+        map.setOnMarkerClickListener { marker ->
+            Timber.d("Marker clicked: %s", marker.title)
+            activity?.let { activity ->
+                activity.supportFragmentManager.beginTransaction()
+                    .replace(R.id.container, DetailsFragment.newInstance(marker.title!!))
```

Comment:
Would be better if you'd used the [Navigation Component](https://developer.android.com/guide/navigation/navigation-getting-started) here.